### PR TITLE
Implement new find functions

### PR DESCRIPTION
**Pull request by PrinceCatt**
1. Add find visit by ic, find patient&visit by medication/diagnosis
2. Updated FileHandler to save and retrieve visits
3. Aligned time format
4. Fixed bug in visit constructor

Notes: 
functional codes should be working, haven't updated the help command
new command: findVisit, findMedication, findDiagnosis

### CODE REVIEW

File: src/main/java/bookbob/functions/FindVisit.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    //find visit by nric and print all visits to terminal
+    public static void findVisitByIc(String nric, Records records) {
+        List<Patient> patientList = records.getPatients();
+        boolean found = false;
+        for (Patient patient : patientList) {
+            if (patient.getNric().equals(nric)) {
+                List<Visit> visits = patient.getVisit();
+                found = true;
+
+                if (visits == null) {
```
Comment (coraleaf0602) [inline src/main/java/bookbob/functions/FindVisit.java:20]: This condition will not be possible as visits is a mandatory field when creating a Patient objects. Hence, visits will always have at least a size of 1

File: src/main/java/bookbob/functions/FindVisit.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        if (!found) {
+            System.out.println("No patient visit record found with NRIC: " + nric);
+        }
+    }
+
+    //find patient by diagnosis and print the specific patient and visit to terminal
+    public static void findVisitByDiagnosis(String symptom, Records records) {
+        List<Patient> patientList = records.getPatients();
+        boolean found = false;
```
Comment (coraleaf0602) [inline src/main/java/bookbob/functions/FindVisit.java:37]: Perhaps change the naming of this boolean variable to isFound as per the Java coding conventions

Comment (PrinceCatt) [reply src/main/java/bookbob/functions/FindVisit.java:37]: noted

File: src/main/java/bookbob/functions/FindVisit.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.List;
+
+public class FindVisit {
+
+    //find visit by nric and print all visits to terminal
+    public static void findVisitByIc(String nric, Records records) {
+        List<Patient> patientList = records.getPatients();
+        boolean isFound = false;
+        for (Patient patient : patientList) {
+            if (patient.getNric().equals(nric)) {
```
Comment (yentheng0110) [inline src/main/java/bookbob/functions/FindVisit.java:16]: Perhaps can use equalsIgnoreCase instead of equals

**Review by yentheng0110 — COMMENTED** [review]
The code LGTM in general!

**Pull request merged**